### PR TITLE
add log error for ping exception

### PR DESCRIPTION
**Pull request by yangruihan**
#60

### CODE REVIEW

**Comment by yangruihan** [comment]
I’m very happy to hear that, but I think there is still room for improvement here. ;)

So I add try catch for ping call, and print some useful message for debug. Otherwise, due to the outer try catch, information will be lost, making it difficult to find the true problem.

File: src/wechaty_puppet_service/utils.py [hunk 1]
```diff
@@ -58,7 +58,14 @@ def test_endpoint(end_point: str, log: Logger) -> int:
...
-        if ping(host) is False:
+        try:
+            if ping(host) is False:
+                res = False
+        except PermissionError as pe:
+            log.error(pe)
+            res = False
+        except errors.PingError as e:
+            log.error(f'got a ping error:\n{e}')
             res = False
```
Comment (wj-Mcat) [inline src/wechaty_puppet_service/utils.py:69]: Your improvement look so great to me. But I don't in what situation the error will be raised.  Can add some more detailed test case for `PermissionError` and `PingError` to help us understand the code ?  If you can, it's great docs for developer to understand more about `endpoint`. How do you think about it ?

**Review by wj-Mcat — CHANGES_REQUESTED** [review]
Your improvement looks so great, but we need your more unit test case.

File: src/wechaty_puppet_service/utils.py (hunk 1 above)
Comment (yangruihan) [reply src/wechaty_puppet_service/utils.py:82]: Thanks for replay. :-)

There will be a `PermissionError`, when I set `end_point` to a service name and run script on a Linux machine without root. And without above codes, due to the outer try catch, I only got `Wechaty - ERROR - The network is not good, the bot will try to restart after 60 seconds.`, which make people confusing. There need more message for figuring out the true problem, so I add try catch for ping call and print possible exception message.

By the way, I optimized the comment.

**Comment by wj-Mcat** [comment]
Yep, I think it's time to merge your code after you bump the version in `VERSION` file to trigger CD stage. So waiting for your new commit.

**Comment by yangruihan** [comment]
Glad to hear this.

I have submitted the `VERSION` file.  :-)

**Comment by wj-Mcat** [comment]
Thank you very much for your contribution!

## You are welcome to join [Wechaty Contributor Program](https://wechaty.js.org/docs/contributor-program/)

### 1. Join Wechaty Organization

> You've invited AIAmber to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

I have invited you to join our Wechaty GitHub Organization, please accept it by following the above message. (See also: [wechaty/PMC#16](https://github.com/wechaty/PMC/issues/16))

### 2. Update Your Wechaty Contributor Profile

1. Please open [Contributor Hall of Fame](https://docs.google.com/spreadsheets/d/1XcDoIczyIclqXP1p90Sz7S0n4Q22xjFzJmjPFlU2g1E/edit#gid=0) and add yourself to the end of the list, so that other contributors will know you better!
1. Please add yourself to our Website [Contributors](https://wechaty.js.org/contributors/) by [creating a PR](https://github.com/wechaty/wechaty.js.org/tree/master/jekyll/_contributors) and refer to this PR link as well.

### 3. Join The Contributor Only WeChat Room

We also have a WeChat room for contributors only which can discuss Wechaty at a deeper level, you are welcome to join and if you are interested.

Please add @lijiarui wechat: _ruirui_0914_ and send her this pr link. She will invite you into Wechaty Contributor Room

Cheers!

**Comment by huan** [comment]
> You've invited Ryan to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

**Pull request merged**